### PR TITLE
[#72] Configure hintman for Heroku deployment

### LINKED ISSUE #72: Figure out how to deploy Hintman
GitHub tutorial briefly mentions to use Heroku.

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #72 

This PR contains the necessary code changes to be able to deploy the project with Heroku, including:

1. Read port from `$PORT` environment variable.
2. Submit `Procfile` with the executable command to launch.

I'm going to test this PR on Heroku first (it's possible to specify custom branch there) and report back whether it works. I'm using the following blog post for the reference:

* https://www.fosskers.ca/blog/deploying-haskell-en.html

### CODE REVIEW

**Comment by chshersh** [comment]
@vrom911 Thanks! Unfortunately, it doesn't work at the moment :disappointed: It takes too much time to build and fails with time limit exceeded error.

**Comment by chshersh** [comment]
Good news everyone! I managed to make it build and it launches! However, you can see how ugly is the way for doing this... And I also need to figure out how to specify secure credentials there. But one step at a time...

File: hintman.cabal [hunk 1]
```diff
@@ -84,10 +84,10 @@ library
 --                         Hintman.Suggestion.TrailingNewline
 --                         Hintman.Suggestion.TrailingSpaces
```
Comment (vrom911) [inline hintman.cabal]: Sorry, but why everything is commented out?

Would that mean that some of the previous PR could be completely wrong or never build?

File: src/Prelude.hs [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
-       , module Json
-       , module Time
-       , module Web
+        , module Colog
+        , module Json
+        , module Time
+        , module Web
 
-       , WithLog
+        , WithLog
```
Comment (vrom911) [inline src/Prelude.hs]: :thinking:

Comment (chshersh) [reply src/Prelude.hs]: Ah, sorry for that, forgot to remove some redundant formatting when I was commenting lines...

**Pull request merged**